### PR TITLE
Weather Refactor

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

Refactors the weather system to no longer rely on effects to deal damage or have weather messages, also improves the ambience code so hopefully it doesn't blast itself to death. All-in-all this should lead to less lag when weather is used on maps.

Additionally, added unique weather events to Big Red, which have a fairly low chance of triggering every twenty minutes. Adds rainy weather to LV-624 as well.

Lastly, reworked Faction Clash to use the weather system rather than manually placing rain sprites.

## Why It's Good For The Game

Weather is cool, and spice up the gameplay loop with dynamic effects. Weather being laggy sucks.

## Changelog

:cl: Geeves, Triiodine
add: Added unique weather events to Big Red / Solaris Ridge and LV-624.
refactor: Refactored weather code to hopefully be less laggy.
refactor: Refactored Faction Clash to use the weather subsystem rather than manually placing rain.
/:cl:

### CODE REVIEW

**Comment by fira** [comment]
I don't see any particular issue but this shoudl probably be TM'ed on the relevant maps

**Comment by Geevies** [comment]
> I don't see any particular issue but this shoudl probably be TM'ed on the relevant maps

this has been TM'd a couple times on both BR and LV with no issues, but I agree we should probably try it out on the new server as well

**Comment by fira** [comment]
Oh, well that should be all good if it's already been tested on old

File: code/controllers/subsystem/weather.dm [hunk 1]
```diff
@@ -80,32 +76,37 @@ SUBSYSTEM_DEF(weather)
...
+	if (last_event_end_time + map_holder.min_time_between_events > world.time || last_event_check_time + map_holder.min_time_between_checks > world.time)
 		return
 
 	// Each map decides its own logic for implementing weather events.
-	if (!is_weather_event_starting && map_holder.should_start_event())
-		// Set up controller state
-		is_weather_event_starting = TRUE
-		weather_event_type = map_holder.get_new_event()
+	if (!is_weather_event_starting)
+		last_event_check_time = world.time
```
Comment (stanalbatross) [inline code/controllers/subsystem/weather.dm:100]: something something COOLDOWN_DECLARE

File: code/datums/weather/weather_event.dm [hunk 2]
```diff
@@ -1,18 +1,47 @@
...
+	var/lightning_chance = 0
+
+/datum/weather_event/proc/start_weather_event()
+	return
+
+// remember, this happens every five seconds or so
+/datum/weather_event/proc/handle_weather_process()
+	if(lightning_chance && prob(lightning_chance))
+		playsound_z(SSmapping.levels_by_trait(ZTRAIT_GROUND), pick('sound/soundscape/thunderclap1.ogg', 'sound/soundscape/thunderclap2.ogg'))
+		for(var/mob/M as anything in GLOB.mob_list)
```
Comment (stanalbatross) [inline code/datums/weather/weather_event.dm:34]: varname

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Pull request merged**